cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -7,130 +7,124 @@
                                 'time', float)
 
+    @sanity_function
+    def assert_reference(self):
+        force = sn.extractsingle(r'1 F=\s+(?P<result>\S+)',
+                                 self.stdout, 'result', float)
+        return sn.assert_reference(force, -.85026214E+03, -1e-5, 1e-5)
 
-@rfm.simple_test
-class VASPCpuCheck(VASPCheck):
-    valid_systems = ['daint:mc', 'eiger:mc', 'pilatus:mc']
-    use_multithreading = False
 
     @run_after('init')
-    def setup_by_scale(self):
-        self.descr = (f'VASP CPU check (version: {self.scale}, '
-                      f'{self.variant})')
-        if self.scale == 'small':
-            self.valid_systems += ['dom:mc']
-            if self.current_system.name in ['daint', 'dom']:
-                self.num_tasks = 12
-                self.num_tasks_per_node = 2
-                self.num_cpus_per_task = 18
-            elif self.current_system.name in ['eiger', 'pilatus']:
-                self.num_tasks = 64
-                self.num_tasks_per_node = 4
-                self.num_cpus_per_task = 8
-                self.num_tasks_per_core = 1
-                self.variables = {
-                    'MPICH_OFI_STARTUP_CONNECT': '1',
-                    'OMP_NUM_THREADS': str(self.num_cpus_per_task),
-                    'OMP_PLACES': 'cores',
-                    'OMP_PROC_BIND': 'close'
-                }
-        else:
-            if self.current_system.name in ['daint']:
-                self.num_tasks = 32
-                self.num_tasks_per_node = 2
-                self.num_cpus_per_task = 18
-
-    @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:mc': {'time': (65.0, None, 0.15, 's')},
-                'daint:mc': {'time': (65.0, None, 0.15, 's')},
-                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
-                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
-            },
-            'large': {
-                'daint:mc': {'time': (55.0, None, 0.15, 's')},
-                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
-                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
-            }
+    def setup_system_filtering(self):
+        self.descr += f' ({self.num_nodes} node(s))'
+
+        # setup system filter
+        valid_systems = {
+            6: ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                'eiger:mc', 'pilatus:mc'],
+            16: ['daint:gpu', 'daint:mc', 'eiger:mc']
         }
-        self.reference = references[self.scale]
 
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
-
-    @run_before('run')
-    def set_cpu_binding(self):
-        self.job.launcher.options = ['--cpu-bind=cores']
+        try:
+            self.valid_systems = valid_systems[self.num_nodes]
+        except KeyError:
+            self.valid_systems = []
```

Comment:
Perhaps it's better if we use `skip_if` here with a message. This will silently ignore the test:

```suggestion
        skip_if(self.num_nodes not in valid_systems, f'no valid systems found for {self.num_nodes}(s)')
        self.valid_systems = valid_systems[self.num_nodes]
```

Reply:
I have added the call to `self.skip_if` as suggested